### ISSUE TITLE
use gatsby-source-mysql with gatsby-image

### ISSUE THREAD

**Issue by guilhermelMoraes** [malcolm-kee/gatsby-source-mysql#7]
Hi! Thank you very much for the plugin. I'm really appreciating it.
It would be very awesome if I could use the gatsby-image plugin with the images that i'm pulling from my database. I'm brand new to development at all, so I don't know if it is already possible and i don't have the knowledge or if isn't really possible...yet. 
I can provide more details about how I wanna use this feature, if you want. 

That's it. Thank you again!

**Comment by malcolm-kee** [comment]
Hi @guilhermelMoraes thanks for raising the issue.
That's an interesting use case that I didn't thought of.
Let me make some exploration because get back to you.

**Comment by malcolm-kee** [comment]
Hi @guilhermelMoraes just to let you know I'm still looking into this feature. It seems more complex than I first thought because it involves saving the file in the cache before `gatsby-image` could process it.

I'm currently checking with other gatsby maintainers if such thing had been done before.

Will keep you updated. Thanks for your patience!

**Comment by guilhermelMoraes** [comment]
Hi @malcolm-kee , thank you in advance for the help and good will, man.  
I've been searching too if there's another way to do that, but unsuccessfully  until now. 
If I realize that there's another way I'll certainly share with you. ;) Thanks again!

**Comment by malcolm-kee** [comment]
Great news! This feature will be possible once the following issue is resolved: https://github.com/gatsbyjs/gatsby/issues/14538
Based on the conversation, that issue will be resolved around 1 week or so.
Then I can work on enhancing this plugin to support image processing.

**Comment by malcolm-kee** [comment]
@guilhermelMoraes While waiting for the PR to be merged, I've made some POC by copy pasting code from `gatsby-source-filesystem` and putting it in `image-processing` branch now.

Could you install it and let me know if it fits your requirement? This would greatly help the turnover time once the dependent PR is merged.

## How to install

Use the following commands to install from git:

```
npm install git://github.com/malcolm-kee/gatsby-source-mysql.git#image-processing
```
OR 
```
yarn add git://github.com/malcolm-kee/gatsby-source-mysql.git#image-processing
```

## Docs

Note that the you need to specify the image columns in your plugin options.

You can refer the updated docs [here](https://github.com/malcolm-kee/gatsby-source-mysql/blob/image-processing/README.md#process-image-saved-as-blob).

**Comment by guilhermelMoraes** [comment]
Hi, @malcolm-kee, i've made what you suggested and isn't working for me. But it's not a problem with the tool itself or the configs. Important to notice:  I also have images that comes from URLs, so in this cases the column on the db it's not the image itself , but its especific URL. 
E.g.: Column -> URL's
data -> https://static.com/static/images/image01.jpg

Thanks in advance for all the help, man.

**Comment by malcolm-kee** [comment]
Ah so it's an URL. I could make that works, but need to change the code further. Will let you know once the change is ready. Thanks for helping to test the change!.

**Comment by malcolm-kee** [comment]
@guilhermelMoraes please reinstall and change your config based on docs [here](https://github.com/malcolm-kee/gatsby-source-mysql/tree/image-processing#process-image-in-remote-url).

Let me know if it works for you.

Note that the API is not finalize and I still thinking what's the best approach.

**Comment by malcolm-kee** [comment]
@guilhermelMoraes try use v2.2.0 branch instead, which allows you to download image from remote url and optimize the image.

```
npm install git://github.com/malcolm-kee/gatsby-source-mysql.git#v2.2.0
```
OR 
```
yarn add git://github.com/malcolm-kee/gatsby-source-mysql.git#v2.2.0
```

Docs [here](https://github.com/malcolm-kee/gatsby-source-mysql/tree/v2.2.0#download-image-for-image-processing).

---

If this fits your use case, I'll publish this version first because the other capability (process image blob) depends on the PR on `gatsby-source-filesystem`.

**Comment by guilhermelMoraes** [comment]
Hi, @malcolm-kee, i've been trying to use the v.2.2.0 branch as you told since the day you post it, but I keep receiving errors.  When I  try to start the development server, it stops on the "source and transform nodes" step. 
I tryied to use the "gatsby clean" to delete the .cache and public folders to test, but it was unsuccessfully. Back to the v.2.1.0 it's working fine

**Comment by malcolm-kee** [comment]
Hi @guilhermelMoraes thanks for letting me know. I need to clarify:

1. when you say "it stops", means the command error out or it's stuck without more message?
2. does it happens the first time you run or it happens after you run it after a few times?
3. I suspect it could be due to the large amount of image that need to be downloaded. Could you try to add `LIMIT` to your query and see it still happens?

Thanks!

**Comment by guilhermelMoraes** [comment]
Hi, @malcolm-kee, thanks for your patience and help. I handled my problem and have used the gatsby-source-mysql with the solution you provided. The problem I've reported above was a problem with another source-plugin. 
Just for clarification: the source and transform nodes got stucked without more messages before I realize the problem was with another plugin. 
Anyway, the v.2.2.0 is working fine. I've been using it for the last 10 days with no problems. When we finish the project we're working for, I'm going to send you a link with the results, if you want to see.
